Assets/Scripts/NetworkController.cs

```diff
@@ -0,0 +1,209 @@
+        ClearRoomListView();
+        UpdateCachedRoomList(roomList);
+        UpdateRoomListView();
+    }
+
+	public override void OnJoinedRoom() {
+		ConnectionStatusText.text = "In Room";
+		LobbyInfoPanel.SetActive(false);
+		RoomInfoPanel.SetActive(true);
+
+		 if (playerListEntries == null) {
+            playerListEntries = new Dictionary<int, GameObject>();
+        }
+
+		// Create and add PlayerListEntryPrefabs for every player in the room to scrollview
+		foreach (Photon.Realtime.Player p in PhotonNetwork.PlayerList) {
+			GameObject entry = Instantiate(PlayerListEntryPrefab);
+			entry.transform.SetParent(RoomInfoContent.transform);
+			entry.transform.localScale = Vector3.one;
+			entry.GetComponent<PlayerListEntry>().Initialize(p.ActorNumber, p.NickName);
+			playerListEntries.Add(p.ActorNumber, entry);
+		}
+	}
+
+	public override void OnPlayerEnteredRoom(Photon.Realtime.Player newPlayer) {
+		// Add new player to scrollview list
+	    GameObject entry = Instantiate(PlayerListEntryPrefab);
+	    entry.transform.SetParent(RoomInfoContent.transform);
+	    entry.transform.localScale = Vector3.one;
+		entry.GetComponent<PlayerListEntry>().Initialize(newPlayer.ActorNumber, newPlayer.NickName);
+		playerListEntries.Add(newPlayer.ActorNumber, entry);
+		//StartGameButton.gameObject.SetActive(CheckPlayersReady());
+	}
+
+	public override void OnLeftRoom() {
+		foreach (GameObject entry in playerListEntries.Values) {
+            Destroy(entry.gameObject);
+        }
+
+        playerListEntries.Clear();
+        playerListEntries = null;
+	}
+
+	public override void OnPlayerLeftRoom(Photon.Realtime.Player otherPlayer) {
+		Destroy(playerListEntries[otherPlayer.ActorNumber].gameObject);
+        playerListEntries.Remove(otherPlayer.ActorNumber);
+
+        //StartGameButton.gameObject.SetActive(CheckPlayersReady());
+	}
+
+	void OnJoinedRoomFailed(short returnCode, string message) {
+		Debug.Log("Joined Room Failed: " + message);
+	}
+
+	public override void OnCreateRoomFailed(short returnCode, string message) {
+		Debug.Log("Create Room Failed: " + message);
+	}	
+
+	#endregion
+
+	public void OnCreateOrJoinRoomButtonClicked() {
+		setLocalPlayerName();
+
+		string roomName = RoomNameInputField.text;
+	    roomName = (roomName.Equals(string.Empty)) ? "Room " + Random.Range(1000, 10000) : roomName;
+		//RoomOptions options = new RoomOptions {MaxPlayers = 2};
+
+		RoomOptions roomOptions = new RoomOptions {};
+		PhotonNetwork.JoinOrCreateRoom(roomName, roomOptions, TypedLobby.Default);
+	}
+
+	public void setLocalPlayerName() {
+		string playerName = PlayerNameInput.text;
+		playerName = (playerName.Equals(string.Empty)) ? "Player " + Random.Range(1000, 10000) : playerName;
+		PlayerNameInput.text = playerName;
+		PhotonNetwork.LocalPlayer.NickName = playerName;
+		Debug.Log("Player name " + playerName);
+		Debug.Log("Photon.LocalPlayer.NickName " + PhotonNetwork.LocalPlayer.NickName);

```

Comment:
Do you need both of these loggers? 
Also some of the indentations in this file seem off

Reply:
noted